CONTRIBUTING.md

```diff
@@ -18,7 +18,119 @@ These are general and loose rules components in UI Kit should strive to fulfill.
 - Documentation of prop-types and usage patterns
 - Sensible defaults for props which are not a hard requirement
 
-### How to add a component to UI Kit.
+## Development
+
+### Testing
+
+We split our testing in three parts: functionality, visual and the final bundle. We use separate tools for these, and we try to avoid overlap.
+
+#### Functionality testing
+
+- write tests from a user perspective, not from a dev perspective
+- we test using `react-testing-library`
+  - we have `test-utils` as a thin wrapper which sets up the context of tested components
+  - we avoid `beforeEach` as it leads to test coupling
+  - an `it` can describe the flow of a user
+  - it is okay to have many `expect` statements in an `it`
+  - code comments can help explain the `expect`s better
+- avoid testing for visuals (classnames or the DOM structure)
+  - the user does not care about the DOM structure or the classnames themselves
+  - use visual testing instead (see below)
+- if mocking is required, try to do it at the level the user interacts with the component
+  - Examples
+    - good: mocking network responses
+    - bad: mocking a fetching component
```

Comment:
maybe we can give a more relevant example to UI Kit, since it's unlikely we will ever have network requests in the lib itself 🤔

Reply:
I was trying to come up with a better example too, but I couldn't really. Maybe this is better?

Maybe:
- good: mocking `localStorage`
- bad: mocking a whole component which uses local storage

It's unlikely too. It's more about getting readers in the mindset that "the more tests reflect how a user uses an application, the more confidence they can give you" than about providing an actual example. So we should use mocks at the intersection with a user (DOM, network, local-storage, cookies) instead of mocking implementation parts of the app. Exceptions apply where it's must more convenient (like `ApolloMockProvider`) and worth the tradeoff.